File: mo-system/mo-system-api/src/main/java/ink/ckx/mo/system/api/model/query/UserPageQuery.java
package ink.ckx.mo.system.api.model.query;

import ink.ckx.mo.common.core.base.BasePageQuery;
import ink.ckx.mo.common.web.enums.StatusEnum;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

/**
 * 用户分页Query
 *
 * @author chenkaixin
 */
@Schema(description = "用户分页Query")
@Data
public class UserPageQuery extends BasePageQuery {

    @Schema(description = "关键字(用户名/昵称/手机号)")
    private String keywords;

    @Schema(description = "用户状态")
    private StatusEnum status;

    @Schema(description = "部门ID")
    private Long deptId;
}

File: mo-admin/mo-admin-api/src/main/kotlin/ink/ckx/mo/admin/api/model/query/UserPageQuery.kt
package ink.ckx.mo.admin.api.model.query

import ink.ckx.mo.common.core.base.BasePageQuery
import ink.ckx.mo.common.web.enums.StatusEnum
import io.swagger.v3.oas.annotations.media.Schema

/**
 * 用户分页Query
 *
 * @author chenkaixin
 */
@Schema(description = "用户分页Query")
data class UserPageQuery(

    @Schema(description = "关键字(用户名/昵称/手机号)")
    var keywords: String? = null,

    @Schema(description = "用户状态")
    var status: StatusEnum? = null,

    @Schema(description = "部门ID")
    var deptId: Long? = null,
) : BasePageQuery()